core/vm/interpreter.go

```diff
@@ -131,6 +123,36 @@ func (in *EVMInterpreter) enforceRestrictions(op OpCode, operation operation, st
 			}
 		}
 	}
+	switch op {
```

Comment:
What's the reason we switch from using `newInstructionSet` for restrictions?

Reply:
`newInstructionSet` now includes ALL available opcodes, where before there was a dedicated instruction set for each hardfork. The logic before checked for the nonexistance of a given opcode, and [this line](https://github.com/ethoxy/multi-geth/pull/19/files/be858d0b67dfb139e669e7c2512abf3cd1e10ff3#diff-8f9edb073f33b63dc5736262579687b8R238) would fail on a "nonexistant" code.

Reply:
> Continuing this pattern would have been impossible because it presents a permutation problem when arbitrary opcode sets can be configured.

Does something like this work?

```
instructionSet := newBasicInstructionSet()
if config.IsEIP1052() {
  instructionSet[EXTCODEHASH] = ...
}
```

Reply:
Yea that would work. One of my initial sketches used this pattern, but I moved away from it thinking the current way seemed 'simpler', with a single resource for known opcodes that I thought might be useful for testing, printing, ... but nothing definite.

Happy to move to this pattern, though. Are you thinking it'd be faster since it avoids the `case` switch for each op (and shifts the expense to a single overhead per interpreter)?

Reply:
It may have slight performance difference, but I just think that using the above pattern might be more future-proof -- what if by accident, we define slightly different opcode on different networks with the same bytecode. WDYT?

Reply:
Ah, different opcode(s) behavior across networks with same bytecode... I hadn't considered that. Alright, will refactor.

Reply:
See 257cf96